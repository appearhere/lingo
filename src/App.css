body {
  margin: 0;
}

.app {
  box-sizing: border-box;
  -webkit-font-smoothing: antialiased;
  margin: 0;
  font-family: 'Avenir Next W01', 'Helvetica Neue', 'Helvetica', 'sans-serif';
  color: var(--color-black);
}
