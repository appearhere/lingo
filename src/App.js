import React from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import GA from 'react-ga';

import ScrollToTop from './components/ScrollToTop/ScrollToTop';
import Header from './components/Header/Header';

import Home from './screens/Home/Home';
import Term from './screens/Term/Term';

import GoogleAnalytics from './utils/GoogleAnalytics';

import css from './App.css';

GA.initialize(process.env.REACT_APP_GA_TRACKING_ID, {
  debug: process.env.NODE_ENV === 'development',
});

const App = () => (
  <Router>
    <ScrollToTop>
      <Route path="/" component={GoogleAnalytics} />
      <div className={css.app}>
        <Route exact path="/:department?/:term?" component={Header} />
        <Route exact path="/:department?" component={Home} />
        <Route exact path="/:department/:term" component={Term} />
      </div>
    </ScrollToTop>
  </Router>
);

export default App;
