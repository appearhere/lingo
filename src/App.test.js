import React from 'react';
import renderer from 'react-test-renderer';

import App from './App';

jest.mock('@appearhere/react-stickynode');
jest.mock('react-ga');
jest.mock('./utils/terms');

describe('App component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<App />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
