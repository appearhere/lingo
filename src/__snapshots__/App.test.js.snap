// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `
<div
  className={undefined}
>
  <div
    className=""
  >
    <h1
      className={undefined}
    >
       Here's the Lingo
    </h1>
    <p
      className={undefined}
    >
      Lingo is here to help you understand the terms we all use.
    </p>
    <section>
      <div
        className={undefined}
      />
    </section>
  </div>
</div>
`;
